### PR TITLE
Speed up parsing and stitching

### LINKED ISSUE #67: BoundingBox5D overlap check is slow
The `BoundingBox5D.overlaps` check is not very performant: 
https://github.com/fmi-faim/faim-hcs/blob/b9f1448260ac9c894dd4a2fe4829a440ef5ddba3/src/faim_hcs/stitching/BoundingBox5D.py#L116

Currently we check for each corner point of a 5D box (32 points) if it is inside another box and vice versa. 

This should be changed to return `False` as soon as one dimension does not overlap i.e. if the bounding boxes do not overlap in the time dimension, no overlap will be possible and we can return after the first dimension check.

### PR DESCRIPTION
**Pull request by tibuch**
* Closes #67, #66, #65.
* Fixes a memory leak when using `map_blocks`.

### CODE REVIEW

**Comment by tibuch** [comment]
Ping @jluethi and @fstur: In case you have already started testing I would like to ask you to update to the latest main branch to get these latest changes.

File: src/faim_hcs/hcs/imagexpress/StackAcquisition.py [hunk 1]
```diff
@@ -72,13 +72,11 @@ def _get_z_spacing(self) -> Optional[float]:
...
 
     def _compute_z_spacing(self, files: pd.DataFrame) -> Optional[float]:
-        if "z" in files.columns:
-            channel_with_stack = files[files["z"] == "2"]["channel"].unique()[0]
-            subset = files[files["channel"] == channel_with_stack]
-            subset = subset[subset["well"] == subset["well"].unique()[0]]
-            subset = subset[subset["field"] == subset["field"].unique()[0]]
-        else:
-            return None
+        assert "z" in files.columns, "No z column in files DataFrame."
```
Comment (imagejan) [inline src/faim_hcs/hcs/imagexpress/StackAcquisition.py:75]: Should we better `raise` instead of `assert` here?

**Review by imagejan — COMMENTED** [review]
Nice improvements, @tibuch 🎉
I'm looking forward to testing these in real use cases now.

File: src/faim_hcs/hcs/imagexpress/StackAcquisition.py (hunk 1 above)
Comment (tibuch) [reply src/faim_hcs/hcs/imagexpress/StackAcquisition.py:75]: I don't have a strong opinion on that. We can replace it with `raise`.

File: src/faim_hcs/hcs/converter.py [hunk 2]
```diff
@@ -142,32 +145,50 @@ def run(
...
                     for ch_metadata in plate_acquisition.get_channel_metadata().values()
                 ]
             }
+
         return plate
 
+    @staticmethod
+    def _wait_for(futures):
+        for future in futures:
+            future.result()
```
[the remark this answers is not in the document]
Comment (tibuch) [reply src/faim_hcs/hcs/converter.py:201]: I was not aware of this function! Will give it a try :)

[the remark this answers is not in the document]
Comment (tibuch) [reply src/faim_hcs/hcs/converter.py:201]: Which implementation are you referring to?

[the remark this answers is not in the document]
Comment (imagejan) [reply src/faim_hcs/hcs/converter.py:201]: `from concurrent.futures import as_completed`

I was using it here recently:
https://github.com/fmi-faim/faim-wako-searchfirst/blob/94f682fdd357231c2b065d7321b2488b1c933d90/src/faim_wako_searchfirst/main.py#L67-L71

File: src/faim_hcs/hcs/imagexpress/StackAcquisition.py (hunk 1 above)
Comment (imagejan) [reply src/faim_hcs/hcs/imagexpress/StackAcquisition.py:75]: Assertion are removed upon optimization, that's why I thought it's better style to use `if` or `try/except` and `raise`..

Comment (tibuch) [reply src/faim_hcs/hcs/imagexpress/StackAcquisition.py:75]: Makes sense!

File: src/faim_hcs/hcs/converter.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (tibuch) [reply src/faim_hcs/hcs/converter.py:201]: I tried
```
as_completed(futures)
```
and
```
list(as_completed(futures))
```

The first one does not write all the data and the second one fails with an `AttributeError`. These are dask futures and might not be exactly implemented the same.

[the remark this answers is not in the document]
Comment (imagejan) [reply src/faim_hcs/hcs/converter.py:201]: Ah, yes, I forgot these aren't the same type of `futures` 🙂 

Looking at the docs (https://docs.dask.org/en/latest/futures.html#waiting-on-futures), I wonder if we can just use:

```
from dask.distributed import wait

wait(futures)
```

**Pull request merged**